accessEntity/data.go

```diff
@@ -0,0 +1,751 @@
+			apiProducts, err = d.getApiProductsByAppName(priVal, "", secVal, "", org)
+		case IdentifierCompanyName:
+			apiProducts, err = d.getApiProductsByAppName(priVal, "", "", secVal, org)
+		case IdentifierApiResource:
+			fallthrough
+		case "":
+			apiProducts, err = d.getApiProductsByAppName(priVal, "", "", "", org)
+		}
+		if err != nil {
+			return
+		}
+	} else if priKey == IdentifierConsumerKey {
+		apiProducts, err = d.getApiProductsByConsumerKey(priVal, org)
+		if err != nil {
+			return
+		}
+	}
+
+	if secKey == IdentifierApiResource {
+		apiProducts = filterApiProductsByResource(apiProducts, secVal)
+	}
+	return
+}
+
+func (d *DbManager) GetApps(org, priKey, priVal, secKey, secVal string) (apps []common.App, err error) {
+	switch priKey {
+	case IdentifierAppId:
+		return d.getAppByAppId(priVal, org)
+	case IdentifierAppName:
+		switch secKey {
+		case IdentifierDeveloperEmail:
+			return d.getAppByAppName(priVal, secVal, "", "", org)
+		case IdentifierDeveloperId:
+			return d.getAppByAppName(priVal, "", secVal, "", org)
+		case IdentifierCompanyName:
+			return d.getAppByAppName(priVal, "", "", secVal, org)
+		case "":
+			return d.getAppByAppName(priVal, "", "", "", org)
+		}
+	case IdentifierConsumerKey:
+		return d.getAppByConsumerKey(priVal, org)
+	}
+	return
+}
+
+func (d *DbManager) GetCompanies(org, priKey, priVal, secKey, secVal string) (companies []common.Company, err error) {
+	switch priKey {
+	case IdentifierAppId:
+		return d.getCompanyByAppId(priVal, org)
+	case IdentifierCompanyName:
+		return d.getCompanyByName(priVal, org)
+	case IdentifierConsumerKey:
+		return d.getCompanyByConsumerKey(priVal, org)
+	}
+	return
+}
+
+func (d *DbManager) GetCompanyDevelopers(org, priKey, priVal, secKey, secVal string) (companyDevelopers []common.CompanyDeveloper, err error) {
+	if priKey == IdentifierCompanyName {
+		return d.getCompanyDeveloperByComName(priVal, org)
+	}
+	return
+}
+
+func (d *DbManager) GetAppCredentials(org, priKey, priVal, secKey, secVal string) (appCredentials []common.AppCredential, err error) {
+
+	switch priKey {
+	case IdentifierConsumerKey:
+		return d.getAppCredentialByConsumerKey(priVal, org)
+	case IdentifierAppId:
+		return d.getAppCredentialByAppId(priVal, org)
+	}
+	return
+}
+
+func (d *DbManager) GetDevelopers(org, priKey, priVal, secKey, secVal string) (developers []common.Developer, err error) {
+	switch priKey {
+	case IdentifierAppId:
+		return d.getDeveloperByAppId(priVal, org)
+	case IdentifierDeveloperEmail:
+		return d.getDeveloperByEmail(priVal, org)
+	case IdentifierConsumerKey:
+		return d.getDeveloperByConsumerKey(priVal, org)
+	case IdentifierDeveloperId:
+		return d.getDeveloperById(priVal, org)
+	}
+	return
+}
+
+func (d *DbManager) getApiProductsByName(apiProdName string, org string) (apiProducts []common.ApiProduct, err error) {
+	err = d.GetDb().QueryStructs(&apiProducts,
+		sql_select_api_product+
+			`WHERE ap.name = ? AND ap.tenant_id IN `+
+			sql_select_tenant_org,
```

Comment:
Can you check and make sure there is no SQL injection issue?
It is ideal to have Prepare & Query
In this case you are just stitching `ap.tenant_id IN` condition, 
potential problem (unless you doing proper input validation while constructing `sql_select_tenant_org`).

Reply:
Good catch.
For `org` this is not a problem because using parameterized query interface (the question mark `?` in sql) can prevent sql injection.
However, for other sql assembling logics, sql injection may be a problem.
i'll make use of `?` in all cases to prevent sql injection.
thanks

Reply:
I still see same code, can you check this again?

Reply:
As I commented, using `?` and arg list in `Query` can prevent SQL injection.
This line is not changed because this line was already secure. I've updated the insecure code.
Please check this commit for details:
https://github.com/apid/apidVerifyApiKey/pull/27/commits/7bcbe27957072ab3724ac050a7395c4850863af0